pgxn/neon/pagestore_smgr.c

```diff
@@ -1565,9 +1730,9 @@ neon_read_at_lsn(RelFileNode rnode, ForkNumber forkNum, BlockNumber blkno,
 
 	if (entry != NULL)
 	{
-		if (entry->slot->effective_request_lsn >= prefetch_lsn)
+		slot = entry->slot;
+		if (slot->effective_request_lsn >= prefetch_lsn)
```

Comment:
May be I missed something but why we are comparing here with `prefetch_lsn` and not with `request_lsn` ?